generators/dynamic.py

```diff
@@ -0,0 +1,169 @@
+import argparse
+import datetime
+import logging
+import random
+
+import pandas
+from dateutil import parser
+
+
+def load(filepath, names):
+    """
+    Loads users from a file
+    :param filepath: path to a file
+    :param names: list that contains column names
+    :return: pandas DataFrame with users
+    """
+    logging.info(f"Started loading {filepath}")
+    data = pandas.read_csv(filepath, names=names)
+    logging.info(f"Finished loading {filepath}")
+    return data
+
+
+def generate_random_timedelta(min_seconds_delta, max_seconds_delta, min_minutes_delta, max_minutes_delta):
+    """
+    Generates random time delta
+    :param min_seconds_delta: lowest possible amount of seconds
+    :param max_seconds_delta: biggest possible amount of seconds
+    :param min_minutes_delta: lowest possible amount of minutes
+    :param max_minutes_delta: biggest possible amount of minutest
+    :return: time delta
+    """
+    return datetime.timedelta(minutes=random.randint(min_minutes_delta, max_minutes_delta),
+                              seconds=random.randint(min_seconds_delta, max_seconds_delta))
+
+
+def get_current_time_with_random_delta(min_seconds_delta, max_seconds_delta, min_minutes_delta, max_minutes_delta):
+    """
+    Takes current time and applies some delta on it
+    :param min_seconds_delta: lowest possible amount of seconds
+    :param max_seconds_delta: biggest possible amount of seconds
+    :param min_minutes_delta: lowest possible amount of minutes
+    :param max_minutes_delta: biggest possible amount of minutest
+
+    :return: current time moved on some time delta
+    """
+    return datetime.datetime.now() + \
+           generate_random_timedelta(min_seconds_delta,
+                                     max_seconds_delta,
+                                     min_minutes_delta,
+                                     max_minutes_delta) \
+           * (1 if random.randint(0, 1) == 1 else -1)
+
+
+def generate_views(users, items, views_number, users_fraction, item_fraction, min_seconds_delta, max_seconds_delta,
+                   min_minutes_delta, max_minutes_delta):
+    """
+    Generates table of views
+    :param item_fraction: fraction of users that will be in views and purchases generation
+    :param users_fraction: fraction of items that will be in views and purchases generation
+    :param users: pandas DataFrame that contains all users
+    :param items: pandas DataFrame that contains all items
+    :param views_number: number of views to generate
+    :return: pandass DataFrame that contains views
+    """
+    logging.info("Started generating views")
+
+    view = users.sample(frac=users_fraction) \
```

Comment:
Don't need to write all stuff in one row. It is dis=fficult for code reading. Split into two variables, first join second sample